src/mame/drivers/dvk_ksm.cpp

```diff
@@ -206,15 +209,28 @@ static INPUT_PORTS_START( ksm )
 	PORT_DIPSETTING(0x0E, "75")
 INPUT_PORTS_END
 
+void ksm_state::device_timer(emu_timer &timer, device_timer_id id, int param, void *ptr)
+{
+	brg_state = !brg_state;
+	m_i8251line->write_txc(brg_state);
+	m_i8251line->write_rxc(brg_state);
+}
+
 void ksm_state::machine_reset()
 {
 	memset(&m_video, 0, sizeof(m_video));
+	brga = 0;
+	brgb = 0;
+	brgc = 0;
+	brg_state = 0;
 }
 
 void ksm_state::video_start()
 {
 	m_tmpclip = rectangle(0, KSM_DISP_HORZ - 1, 0, KSM_DISP_VERT - 1);
 	m_tmpbmp.allocate(KSM_DISP_HORZ, KSM_DISP_VERT);
+
+	m_brg = timer_alloc(0);
```

Comment:
If you're going to use this form of the timer, you should really use a named constant and check for it to make it a bit easier to play nice with device interfaces and derived classes.